packages/just-scripts-utils/src/findMonoRepoRootPath.ts

```diff
@@ -3,7 +3,12 @@ import path from 'path';
 import fse from 'fs-extra';
 import { readPackageJson } from './readPackageJson';
 
-export function findMonoRepoRootPath() {
+/**
+ * Find the path of a monorepo root relative to the project being generated/updated.
+ * This will be either the directory containing rush.json or the root of a package which uses
+ * the `just-stack-monorepo` stack according to the `just.stack` property of its package.json.
+ */
+export function findMonoRepoRootPath(): string | null {
```

Comment:
Wondering if we should make these utility functions to be decoupled from the `paths` thing and have that be passed in as an arg. Maybe future refactoring.

Reply:
Yeah using an arg (or having an optional override in an arg) would make it a bit clearer what path it's operating on.

Reply:
I can't even tell where `paths.projectPath` is getting set currently for `addPackageTask` and `upgradeRepoTask`...which feels like using magical goo to tie things together. Are they just using the default (`process.cwd()`) right now, or is it assuming the task consumer sets `paths.projectPath` if they want to override?

Reply:
Updated to take the project path as a parameter. Let me know if the way I did it looks okay.

Reply:
@kenotron Now the build fails because a task function (such as `addPackageTask` or `upgradeRepoTask`) is expected to have a `done` callback and no other parameters. Any suggestions for how to reconcile this?

```ts
export interface TaskFunction extends Undertaker.TaskFunctionParams {
  (this: TaskContext, done: (error?: any) => void): void | Duplex | NodeJS.Process | Promise<never> | any;
}
```

Reply:
I reverted the project path as param thing for now to unblock this, and we can come back to it later.